src/main/sensors/voltage.c

```diff
@@ -194,7 +228,12 @@ void voltageMeterADCInit(void)
         voltageMeterADCState_t *state = &voltageMeterADCStates[i];
         memset(state, 0, sizeof(voltageMeterADCState_t));
 
-        pt1FilterInit(&state->filter, pt1FilterGain(GET_BATTERY_LPF_FREQUENCY(batteryConfig()->vbatLpfPeriod), HZ_TO_INTERVAL(50)));
+        pt1FilterInit(&state->displayFilter, pt1FilterGain(GET_BATTERY_LPF_FREQUENCY(batteryConfig()->vbatDisplayLpfPeriod), HZ_TO_INTERVAL(getBatteryVoltageTaskFrequencyHz())));
+#if defined(USE_BATTERY_VOLTAGE_SAG_COMPENSATION)
+        if (currentPidProfile->vbat_sag_compensation > 0) {
+            pt1FilterInit(&state->sagFilter, pt1FilterGain(GET_BATTERY_LPF_FREQUENCY(batteryConfig()->vbatSagLpfPeriod), HZ_TO_INTERVAL(getBatteryVoltageTaskFrequencyHz())));
+        }
```

Comment:
This needs to be changed to check for any profile having the feature enabled, not just the current profile. Otherwise if the board boots with the feature disabled in the active profile then subsequently changes to a profile with the feature enabled, the filter will be in an uninitialized and nonfunctional state.